### PR TITLE
Allow bidirectional communication for TouchOsc

### LINKED ISSUE #313: can't get feedback to work with Touch OSC
I'm trying to use Touch OSC through Modality Toolkit.
I'm running SC 3.9.3 and have installed Modality through `Quarks.gui`
On my iPad running Touch OSC 1.9.10, I use the Simple Layout.

The following code works :

```supercollider
t = MKtl(\tosca, "touchosc-simple1");
t.device.updateSrcAddr("192.168.1.12", 63763); // IP and port of TouchOsc on my iPad as printed
t.gui;
t.elAt.action = { |el|
    "tOsc %: %\n".postf(el.name, el.value.round(0.001))
};
```

I'm getting values printed on the console when I move a slider on the iPad, like

`tOsc sl_1: 0.316`

Great.

But how do I set a value on SC side and have the slider move on the iPad ?

I tried

```supercollider
t.setValueAt(\sl_1, 0.0);
```

But the slider does not move on the iPad.
Same if I move a slider on the GUI displayed by t.gui

If I type

```supercollider
OSCFunc.trace(true);
```

and move a slider on the iPad, I get the following on the console so IP and port seems to be OK.

```
OSC Message Received:
    time: 104.314042791
    address: a NetAddr(192.168.1.12, 63763)
    recvPort: 57120
    msg: [ /1/fader1, 0.49654316902161 ]
```

Thanks in advance for your help.

### PR DESCRIPTION
**Pull request by geoffroymontel**
# Purpose
Fixes #313 

# Test case

* start TouchOsc on your iPhone, iPad or Android device and load the Simple layout
* note your device local IP address and incoming port in TouchOSC settings
* on Supercollider

```supercollider
t = MKtl(\tosca, "touchosc-simple1");
t.device.updateSrcAddr("192.168.1.12", 63763); // IP and port of TouchOsc on your device
t.elAt.action = { |el|
    "tOsc %: %\n".postf(el.name, el.value.round(0.001))
};
```

* Move the first slider on the left on the device, you should get values on supercollider console such as 

```
tOsc sl_1: 0.316
```

* On Supercollider, set the value of the first slider with

```supercollider
t.setValueAt(\sl_1, rrand(0.0,1.0));
```

the first slider should move on the device.

### CODE REVIEW

**Comment by LFSaw** [comment]
Thanks for this!
I'd prefer a separate method `updateDstAddr` so that it is possible to set source and destination to different values.
Alternatively, you can also already now (without touching the code) do:

```
t = MKtl(\tosca, "touchosc-simple1");

// IP and port of TouchOsc on your device
t.device.destination. hostname = "192.168.1.12"; 
t.device.destination. port = 63763; 

t.elAt.action = { |el|
    "tOsc %: %\n".postf(el.name, el.value.round(0.001))
};
```

**Comment by geoffroymontel** [comment]
Thanks @LFSaw for the review. I will change it to add a separate `updateDstAddr`.
I could not find a use case where src and dest addresses would be different. Do you have an example ?
Thanks !

**Comment by LFSaw** [comment]
If you e.g. use the Pd OSC objects, it is AFAIK not possible to manually set the port from which you're sending.
If this is not desired behaviour (having the possibility to have source and destination different), we should consider refactoring the code accordingly.

**Comment by geoffroymontel** [comment]
Looking at the source code, the `source` parameter is used in the `OSCFunc` to filter incoming OSC messages from the source IP address.

Cf. OSCFun documentation :

```supercollider
OSCFunc.new(func, path, srcID, recvPort, argTemplate, dispatcher)

srcID : An optional instance of NetAddr indicating the **IP address of the sender**. If set this object will only respond to messages from that source.
```

Which means the "port" part of source does not seem to be used at all.

`destination` is used to send back OSC packets to the sender and here, of course, we need both IP and port.

@LFSaw @adcxyz do you think a scenario can exist where source IP is different from destination IP ?
In such a case, I guess merging `source` and `destination` could be a possible refactoring.

**Comment by geoffroymontel** [comment]
@LFSaw I followed your suggestion and reverted back `updateSrcAddr` and added `updateDstAddr` and `updateSrcDstAddr`.

I still can't think of a scenario where source IP <> destination IP, if you can't either and you want me to go over refactoring, I'm in !

**Comment by geoffroymontel** [comment]
Do you think this PR is ready enough for merge ?

File: Modality/MKtlDescriptions/_allDescs.cache.scd [hunk 1]
```diff
@@ -22,7 +22,6 @@ Dictionary[
 	("evolution-ucontrol-uc33" -> "UC-33 USB MIDI Controller"),
 	("faderfox-uc4" -> "Faderfox UC4"),
 	("faderfox-uc4_pg" -> "Faderfox UC4"),
-	("fio-somo" -> "fio-somo"),
```
Comment (LFSaw) [inline Modality/MKtlDescriptions/_allDescs.cache.scd]: why is this file in this PR?

everything else looks good :)

**Comment by LFSaw** [comment]
sorry, AFAICS, now you deleted the file `_allDescs.cache.scd` from the repository altogether... what I meant is that you should not add changes to the file to the PR since it is not related to the theme of this PR... (can be a separate PR if needed, though).

**Comment by LFSaw** [comment]
Thanks for your patience!

**Comment by geoffroymontel** [comment]
Thanks for your detailed review and your patience too !! :)

**Pull request merged**